base/exports.jl

```diff
@@ -950,6 +950,7 @@ export
     sizeof,
 
 # tasks and conditions
+    channeled_tasks,
```

Comment:
I can remove the export for now. Can revisit it later based on usage. For now I'll add a `channel(T, size, function) -> (Channel, Task)` convenience function which creates and binds a channel to a 1-arg (the created channel object) function.